webapp/src/components/ProfilePage/ProfilePage.js

```diff
@@ -0,0 +1,208 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import {
+  Menu,
+  Container,
+  Card,
+  Pagination,
+  Loader,
+  Dimmer,
+  Label
+} from 'semantic-ui-react'
+import AddressLink from 'components/AddressLink'
+import Navbar from 'components/Navbar'
+import Publication from 'components/MarketplacePage/Publication'
+import Parcel from './Parcel'
+import Contribution from './Contribution'
+import { publicationType } from 'components/types'
+
+import { TABS, buildUrl } from './utils'
+
+import './ProfilePage.css'
+
+export default class ProfilePage extends React.PureComponent {
+  static propTypes = {
+    publications: PropTypes.arrayOf(publicationType),
+    page: PropTypes.number.isRequired,
+    pages: PropTypes.number.isRequired,
+    sortBy: PropTypes.string.isRequired,
+    sortOrder: PropTypes.string.isRequired,
+    onNavigate: PropTypes.func.isRequired,
+    onConnect: PropTypes.func.isRequired,
+    onFetchPublications: PropTypes.func.isRequired
+  }
+
+  componentWillMount() {
+    const { onConnect, onFetchAddress } = this.props
+    onConnect()
+    onFetchAddress()
+  }
+
+  handlePageChange = (event, data) => {
+    const { address, tab, onNavigate } = this.props
+    const url = buildUrl({
+      page: data.activePage,
+      address,
+      tab
+    })
+    onNavigate(url)
+  }
+
+  renderLoading() {
+    return (
+      <Dimmer active inverted>
+        <Loader size="huge" />
+      </Dimmer>
+    )
+  }
+
+  renderEmpty() {
+    const { tab } = this.props
+    return (
+      <div className="empty">
+        <p>This user has no {tab}</p>
+      </div>
+    )
+  }
+
+  renderPublications() {
+    const { publications } = this.props
+    return (
+      <Card.Group stackable={true}>
+        {publications.map((publication, index) => (
+          <Publication
+            key={publication.tx_hash}
+            publication={publication}
+            debounce={index * 100}
+          />
+        ))}
+      </Card.Group>
+    )
+  }
+
+  renderGrid() {
+    const { grid, tab } = this.props
+    switch (tab) {
+      case TABS.PARCELS: {
+        return (
+          <Card.Group stackable={true}>
+            {grid.map(parcel => <Parcel key={parcel.id} parcel={parcel} />)}
+          </Card.Group>
+        )
+      }
+      case TABS.CONTRIBUTIONS: {
+        return (
+          <Card.Group stackable={true}>
+            {grid.map(contribution => (
+              <Contribution
+                key={contribution.district_id}
```

Comment:
Sadly `district_id` can be duplicated because an address could've contributed to the same district multiple times for different `land_count`s (and we're not merging those anywhere). Maybe we can use `index` here?

Also, to discuss later, we might want to add the contribution date to the UI.

Reply:
contributions are getting grouped by `district_id`. I added two contributions of 6 LAND to `AETHERIAN project`  and the API returns a single contribution of 12 LAND

Reply:
👉 https://github.com/decentraland/marketplace/blob/master/src/Contribution/Contribution.js#L22